lib/ipv4/ipv4.ml

```diff
@@ -97,61 +97,66 @@ module Make(Ethif: V1_LWT.ETHIF) (Arpv4 : V1_LWT.ARP) = struct
   end
 
   let adjust_output_header ~dmac ~tlen frame =
-    Wire_structs.set_ethernet_dst dmac 0 frame;
-    let buf = Cstruct.sub frame Wire_structs.sizeof_ethernet Wire_structs.Ipv4_wire.sizeof_ipv4 in
+    let open Ipv4_wire in
+    Ethif_wire.set_ethernet_dst dmac 0 frame;
+    let buf = Cstruct.sub frame Ethif_wire.sizeof_ethernet sizeof_ipv4 in
     (* Set the mutable values in the ipv4 header *)
-    Wire_structs.Ipv4_wire.set_ipv4_len buf tlen;
-    Wire_structs.Ipv4_wire.set_ipv4_id buf (Random.int 65535); (* TODO *)
-    Wire_structs.Ipv4_wire.set_ipv4_csum buf 0;
+    set_ipv4_len buf tlen;
+    set_ipv4_id buf (Random.int 65535); (* TODO *)
+    set_ipv4_csum buf 0;
     let checksum = Tcpip_checksum.ones_complement buf in
-    Wire_structs.Ipv4_wire.set_ipv4_csum buf checksum
+    set_ipv4_csum buf checksum
 
   let allocate_frame t ~dst ~proto =
+    let open Ipv4_wire in
     let ethernet_frame = Io_page.to_cstruct (Io_page.get 1) in
```

Comment:
Here, we're trying to allocate "enough" memory for whatever the caller's data might be, without knowing how much that is, so a page seems like a reasonable amount.  Do you think changing it to Cstruct.create 4096 is materially different?

(I'd like to remove this function entirely, and change the "ip allocates a buffer, other modules write into it at the provided offset" to something else, but that's a rather large API change)